crates/oxc_minifier/src/ast_passes/peephole_substitute_alternate_syntax.rs

```diff
@@ -323,6 +330,154 @@ impl<'a> PeepholeSubstituteAlternateSyntax {
             None
         }
     }
+
+    fn compress_new_expression(&mut self, expr: &mut Expression<'a>, ctx: &mut TraverseCtx<'a>) {
+        if let Expression::NewExpression(new_expr) = expr {
+            // `new Object` -> `{}`
+            if new_expr.arguments.is_empty()
+                && new_expr.callee.is_global_reference_name("Object", ctx.symbols())
+            {
+                *expr =
+                    ctx.ast.expression_object(expr.span(), Vec::new_in(ctx.ast.allocator), None);
+                self.changed = true;
+            } else if new_expr.callee.is_global_reference_name("Array", ctx.symbols()) {
+                // `new Array` -> `[]`
+                if new_expr.arguments.is_empty() {
+                    *expr = self.empty_array_literal(ctx);
+                    self.changed = true;
+                } else if new_expr.arguments.len() == 1 {
+                    let Some(arg) = new_expr.arguments[0].as_expression() else { return };
+                    // `new Array(0)` -> `[]`
+                    if arg.is_number_0() {
+                        *expr = self.empty_array_literal(ctx);
+                        self.changed = true;
+                    }
+                    // `new Array(8)` -> `Array(8)`
+                    else if arg.is_number_literal() {
+                        *expr = self.array_constructor_call(
+                            new_expr.arguments.clone_in(ctx.ast.allocator),
+                            ctx,
+                        );
+                        self.changed = true;
```

Comment:
In this context, you can simply use `Some(Expression)` for changing (otherwise `None`) and replace it in `Traverse` impl. This provides a more convenient approach.